ffi/input_pocketbook.lua

```diff
@@ -0,0 +1,147 @@
+local ffi = require("ffi")
+local C = ffi.C
+local inkview = ffi.load("inkview")
+local util = require("ffi/util")
+require("ffi/linux_input_h")
+
+-- luacheck: push
+-- luacheck: ignore
+local EVT_INIT = 21
```

Comment:
I wouldn't mind a reference as to where those magic numbers come from ;). (<inkview.h>?)

Reply:
I was tempted to use cdef generator at first, but ran into issues of inkview versions being utter and complete mess, plus  part of the API being undocumented, ie not in the inkview header at all.

Including it only for the sake of single enum *all the while* awkwardly avoiding the minefield of the version mess stuff of theirs isn't worth the effort imo. The bonus side of this is that since there's no hard version linkage to inkview anymore, the SDK stuff becomes a bit less fragile due to room for ad-hoc fixups in ffi.

Reply:
Okay, that's what I was afraid of, just wanted to make sure ;). :+1:

Reply:
@NiLuJe After a bit of digging, turns out the the header at https://github.com/pocketbook-free/SDK_481/blob/5.12/arm-obreey-linux-gnueabi/sysroot/usr/local/include/inkview.h *seems* somewhat useable - at least the MT struct size seem  corrected now, and there's even function signatures for bit of the stuff that used to be undocumented - will have to test it first to see if it actually runs.

Provided it runs ok with no need to retouch stuff, it might be worth the trouble. The build process will have to commit to using only this recent versions of the sdk to get the header, which may spell ABI issues due to unrelated libraries in the SDK that koreader depends on being too new on too old readers. So it's up to you, generated cdef, at the cost of possibly having to deal with tardiness of the SDK, or ugly code of hardcoded cdefs, with a little more wiggle room build wise.

Reply:
The build-process is entirely irrelevant, the burden to run ffi-cdecl is left to the developer ;).

So, basically, just a .cdecl with a comment about it if there are API issues would be enough (especially as compared to what the TC pulls: https://github.com/koreader/koxtoolchain/blob/2440ef904d88a3d4f8bff98d97df9a61e4afd86f/gen-tc.sh#L171-L172).

(There's already quite a few things that wouldn't quite work if they were simply being auto-generated at compile-time ;)).

Reply:
i.e., basically, it's just a question of traceability more than anything else, in case someone else stumbled onto the cdef block ;).

Reply:
Welp, that was more obtuse than anticipated, but brought the entirety of inkview as ffi-cdecl in the end. Might come handy when frontend or plugin wants to bring in something from it.

Reply:
Yeah, it can be a bit.... awkward to grok ;).